### PR TITLE
CXP-2462: Rework the logic of detecting the test mode in the S3 sink connector

### PR DESCRIPTION
**Pull request by sugarcrm-jgminder**

### CODE REVIEW

File: common/src/main/java/com/spredfast/kafka/connect/s3/S3.java [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
-                  new EndpointConfiguration(config.get("s3.endpoint"), config.get("region")))
-              .withCredentials(
-                  new AWSStaticCredentialsProvider(
-                      new BasicAWSCredentials(
-                          config.get("aws.accessKeyId"), config.get("aws.secretAccessKey"))))
-              .withPathStyleAccessEnabled(s3PathStyle)
-              .build();
+    // If worker config sets explicit endpoint override (e.g. for testing) use that
+    String s3Endpoint = config.get("s3.endpoint");
+    if (s3Endpoint != null && !Objects.equals(s3Endpoint, "")) {
```
Comment (morozov) [inline common/src/main/java/com/spredfast/kafka/connect/s3/S3.java:18]: I wouldn't ignore the empty string. If the default endpoint should be used, the parameter should be completely omitted.

File: common/src/main/java/com/spredfast/kafka/connect/s3/S3.java [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
+      s3Client.setEndpoint(s3Endpoint);
```
Comment (morozov) [inline common/src/main/java/com/spredfast/kafka/connect/s3/S3.java:19]: This method is deprecated. The SDK recommends using the builder as it was done previously.

**Review by morozov — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/spredfast/kafka/connect/s3/S3.java (hunk 1 above)
Comment (sugarcrm-jgminder) [reply common/src/main/java/com/spredfast/kafka/connect/s3/S3.java:18]: I just rolled back this version of the file to it's previous working production version, but this makes sense to me. I'll update.

File: common/src/main/java/com/spredfast/kafka/connect/s3/S3.java (hunk 2 above)
Comment (sugarcrm-jgminder) [reply common/src/main/java/com/spredfast/kafka/connect/s3/S3.java:19]: Upon further inspection, using the new API requires the use of `EndpointConfiguration(s3Endpoint, signingRegion)` which will require the `mothership-operator` to add a region property to the connectors if this is deployed again prior to endpoint removal. (i.e. not backwards compatible). I'd recommend dealing with the deprecation and doing this after the endpoint can be removed from `mothership-operator`.

Comment (morozov) [reply common/src/main/java/com/spredfast/kafka/connect/s3/S3.java:18]: Sounds good. Let's get a ticket filed to transition to the recommended API.

**Review by morozov — APPROVED** [review]

**Pull request merged**